proxy/txnmanager/transaction.go

```diff
@@ -0,0 +1,329 @@
+package txnmanager
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/dedis/d-voting/contracts/evoting"
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+	"go.dedis.ch/dela"
+	"go.dedis.ch/dela/core/execution/native"
+	"go.dedis.ch/dela/core/ordering/cosipbft/blockstore"
+	"go.dedis.ch/dela/core/txn"
+	"go.dedis.ch/dela/core/txn/pool"
+	"go.dedis.ch/dela/crypto"
+	"go.dedis.ch/dela/serde"
+	"go.dedis.ch/kyber/v3"
+	"golang.org/x/xerrors"
+)
+
+const (
+	maxTimeTransactionCheck = 10 * time.Minute
+)
+
+// NewTransactionManager returns a new initialized transaction manager
+func NewTransactionManager(mngr txn.Manager, p pool.Pool,
+	ctx serde.Context, pk kyber.Point, blocks blockstore.BlockStore, signer crypto.Signer) Manager {
+
+	logger := dela.Logger.With().Timestamp().Str("role", "evoting-proxy").Logger()
+
+	return &manager{
+		logger:  logger,
+		context: ctx,
+		mngr:    mngr,
+		pool:    p,
+		pk:      pk,
+		blocks:  blocks,
+		signer:  signer,
+	}
+}
+
+// form defines HTTP handlers to manipulate the evoting smart contract
+//
+// - implements proxy.Transaction
+type manager struct {
+	sync.Mutex
+
+	logger  zerolog.Logger
+	context serde.Context
+	mngr    txn.Manager
+	pool    pool.Pool
+	pk      kyber.Point
+	blocks  blockstore.BlockStore
+	signer  crypto.Signer
+}
+
+// IsTxnIncluded
+// Check if the transaction is included in the blockchain
+func (h *manager) IsTxnIncluded(w http.ResponseWriter, r *http.Request) {
+	// get the token from the url
+	vars := mux.Vars(r)
+
+	// check if the token is valid
+	if vars == nil || vars["token"] == "" {
+		http.Error(w, fmt.Sprintf("token not found: %v", vars), http.StatusInternalServerError)
+		return
+	}
+
+	token := vars["token"]
+
+	// decode the token
+	marshall, err := b64.URLEncoding.DecodeString(token)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to decode token: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	// unmarshall the token to get the json with all the informations
+	var content TransactionInfo
+	err = json.Unmarshal(marshall, &content)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to unmarshall token: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	// check if the transaction status is unknown
+	// if it is not unknown, it means that the transaction was already checked
+	if content.Status != UnknownTransactionStatus {
```

Comment:
you should have a separate function `func (t TransactionInfo) validate() error{...}` that validates the info (status, hash, signature, ...).